src/prefect/utilities/engine.py

```diff
@@ -559,8 +559,12 @@ def set_state_and_handle_waits(set_state_func) -> OrchestrationResult:
         )
 
 
-def _dynamic_key_for_task_run(context: FlowRunContext, task: Task) -> Union[int, str]:
-    if context.detached:  # this task is running on remote infrastructure
+def _dynamic_key_for_task_run(
+    context: FlowRunContext, task: Task, stable: bool = False
```

Comment:
this default looks wrong - subflows need `stable=True` which isn't set anywhere

Reply:
you are most certainly right